src/Yaapii.Zip/ZipWithPassword.cs

```diff
@@ -11,35 +13,59 @@ namespace Yaapii.Zip
     /// </summary>
     public sealed class ZipWithPassword : IInput
     {
-        private readonly IScalar<IInput> zip;
+        private readonly Sticky<IInput> result;
 
         /// <summary>
         /// A Zip which input is saved with a password
         /// </summary>
-        public ZipWithPassword(string name, string password, IInput origin) : this(
-            new Sticky<IInput>(() =>
+        public ZipWithPassword(string password, string name, IInput origin) : this(
```

Comment:
The naming of variables suggests, that this is a decorator, which it is not. Maybe rename them to: ```new ZipWithPassword(string password, string path, IInput content)```